physionet-django/console/views.py

```diff
@@ -1402,6 +1403,75 @@ def guidelines_review(request):
         {'guidelines_review_nav': True})
 
 
+@login_required
+@user_passes_test(is_admin, redirect_field_name='project_home')
+def editorial_stats(request):
+    """
+    Editorial stats for reviewers.
+    """
+    approved_projects = PublishedProject.objects.all().order_by('-publish_datetime')
+    # Filter out legacy projects
+    approved_projects = [i for i in approved_projects if i.editor_assignment_datetime != None]
+
+    # Create a set of years in which projects were published
+    years_published = sorted(
+        list(set([i.publish_datetime.year for i in approved_projects]))
+    )
+    # Determine the number of projects published per year
+    num_published = {}
+    for year in years_published:
+        num_published[str(year)] = 0
+
+    for project in approved_projects:
+        num_published[str(project.publish_datetime.year)] += 1
+
+    num_published = list(num_published.values())
+
+    # Determine the elapsed times between editorial events
+    yearly_sub_ed = []
+    yearly_sub_pub = []
+    for year in years_published:
+        # Collect all the values for each year to be averaged
+        temp_yearly_sub_ed = []
+        temp_yearly_sub_pub = []
+        for project in approved_projects:
+            if (project.publish_datetime.year == year):
+                # Filter out backdated projects
+                if project.editor_assignment_datetime > project.submission_datetime:
+                    temp_yearly_sub_ed.append(
+                        (project.editor_assignment_datetime - \
+                        project.submission_datetime).days
+                    )
+                # Filter out backdated projects
+                if project.publish_datetime > project.submission_datetime:
+                    temp_yearly_sub_pub.append(
+                        (project.publish_datetime - \
+                        project.submission_datetime).days
+                    )
+        try:
+            yearly_sub_ed.append(
+                round(statistics.median(temp_yearly_sub_ed),1)
+            )
+        except statistics.StatisticsError:
+            # If there are no values to take the median of
+            yearly_sub_ed.append(0)
+        try:
+            yearly_sub_pub.append(
+                round(statistics.median(temp_yearly_sub_pub),1)
+            )
+        except statistics.StatisticsError:
+            # If there are no values to take the median of
+            yearly_sub_pub.append(0)
+
+    yearly_stats = zip(years_published,
+                       num_published,
+                       yearly_sub_ed,
+                       yearly_sub_pub)
+
+    return render(request, 'console/editorial.html',
```

Comment:
if the approach suggested above is used, the template would need to be changed slightly:

```
        {% for year, values in stats.items %}
          <tr>
            <td>{{ year }}</td>
            <td>{{ values.0 }}</td>
            <td>{{ values.1 }}</td>
            <td>{{ values.2 }}</td>
```